tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py

```diff
@@ -218,6 +222,101 @@ def test_container_combos_invalid(
                 code_section,
                 first_sub_container,
             ],
+            kind=container_kind,
+        ),
+        expect_exception=EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container,container_kind",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            stop_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/STOP",
+        ),
+        pytest.param(
+            eofcreate_code_section,
+            return_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/RETURN",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            returncontract_sub_container,
+            ContainerKind.INITCODE,
+            id="RETURNCONTRACT/RETURNCONTRACT",
+        ),
+    ],
+)
+def test_container_combos_deeply_nested_invalid(
+    eof_test: EOFTestFiller,
+    code_section: Section,
+    first_sub_container: Container,
+    container_kind: ContainerKind,
+):
+    """Test invalid subcontainer reference / opcode combos on a deep container nesting level"""
+    invalid_container = Container(
+        sections=[
+            code_section,
+            first_sub_container,
+        ],
+        kind=container_kind,
+    )
+
+    container = invalid_container
+    while len(container) < 0x6000:
+        container = Container(
+            sections=[
+                eofcreate_code_section,
+                Section.Container(container=container.copy()),
+            ],
+            kind=ContainerKind.INITCODE,
+        )
+
+    eof_test(
+        data=container,
+        expect_exception=EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container,container_kind",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            stop_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/STOP",
+        ),
+        pytest.param(
+            eofcreate_code_section,
+            return_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/RETURN",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            returncontract_sub_container,
+            ContainerKind.INITCODE,
+            id="RETURNCONTRACT/RETURNCONTRACT",
+        ),
+    ],
+)
+def test_container_combos_non_first_code_sections_invalid(
+    eof_test: EOFTestFiller,
+    code_section: Section,
+    first_sub_container: Container,
+    container_kind: ContainerKind,
+):
+    """Test invalid subcontainer reference / opcode combos in a non-first code section"""
+    eof_test(
+        data=Container(
+            sections=[Section.Code(Op.JUMPF[i]) for i in range(1, 1024)]
+            + [code_section, first_sub_container],
+            kind=container_kind,
```

Comment:
Similar comment to previous one, we should include a valid option in the parametrize to prove that the pattern only fails in case of an incorrect sub-container.

Reply:
Added.